frontends/mit-open/src/components/RestrictedRoute/RestrictedRoute.tsx

```diff
@@ -41,7 +38,9 @@ const RestrictedRoute: React.FC<RestrictedRouteProps> = ({
   children,
   requires,
 }) => {
-  if (!hasPermission(requires)) {
+  const { isLoading, data: user } = useUserMe()
+
```

Comment:
```suggestion
if (isLoading) return null
```
IMO better to show nothing if the userquery is loading.

Since the header loads `useUserMe`, users navigating around the app would probably not trigger the early return. But without it, someone going directly to a restricted page (typing in the URL, following an external link) would get a flash of the page before being redirected.